static/app/views/settings/featureFlags/changeTracking/newProviderForm.tsx

```diff
@@ -37,10 +40,18 @@ type CreateSecretResponse = string;
 
 export default function NewProviderForm({
   onCreatedSecret,
+  onProviderChange,
   onSetProvider,
+  onError,
+  canOverrideProvider,
+  existingSecret,
 }: {
+  canOverrideProvider: boolean;
   onCreatedSecret: (secret: string) => void;
+  onError: (error: string | null) => void;
+  onProviderChange: (provider: string) => void;
   onSetProvider: (provider: string) => void;
+  existingSecret?: Secret;
 }) {
```

Comment:
These props are getting long!

It's a good habit to move props into their own type. Don't need to think of a name each time either, `Props` is great. This makes it easier for the next person to keep adding stuff over time, and if the types get complex it'll be easier to extracting stuff into more named types

```
interface Props {
 canOverrideProvider: boolean;
  onCreatedSecret: (secret: string) => void;
  onError: (error: string | null) => void;
  onProviderChange: (provider: string) => void;
  onSetProvider: (provider: string) => void;
  existingSecret?: Secret;
}

export default function NewProviderForm({
  onCreatedSecret,
  onProviderChange,
  onSetProvider,
  onError,
  canOverrideProvider,
  existingSecret,
}: Props) {
  ...
```